streams/shared/src/main/scala/zio/stream/internal/ChannelExecutor.scala

```diff
@@ -21,7 +21,7 @@ private[zio] class ChannelExecutor[Env, InErr, InElem, InDone, OutErr, OutElem,
     val currInput = input
     input = prev
 
-    if (currInput ne null) currInput.close(exit) else ZIO.unit
+    if (currInput ne null) currInput.close(exit) else Exit.unit
```

Comment:
Minor suggestion, but maybe we should change `addFinalizer` so that it doesn't push the finalizer in the stack in case that it's `eq Exit.unit`